pkg/juicefs/mount/pod_mount.go

```diff
@@ -348,23 +347,26 @@ func (p *PodMount) waitUtilMountReady(jfsSetting *jfsConfig.JfsSetting, podName
 		time.Sleep(time.Millisecond * 500)
 	}
 	// mountpoint not ready, get mount pod log for detail
-	log, err := p.getErrContainerLog(podName)
+	log, err := p.getErrContainerLog(ctx, podName)
 	if err != nil {
 		klog.Errorf("Get pod %s log error %v", podName, err)
 		return fmt.Errorf("mount %v at %v failed: mount isn't ready in 30 seconds", util.StripPasswd(jfsSetting.Source), jfsSetting.MountPath)
 	}
 	return fmt.Errorf("mount %v at %v failed: %v", util.StripPasswd(jfsSetting.Source), jfsSetting.MountPath, log)
 }
 
-func (p *PodMount) waitUtilJobCompleted(jobName string) error {
+func (p *PodMount) waitUtilJobCompleted(ctx context.Context, jobName string) error {
 	// Wait until the job is completed
-	for i := 0; i < 120; i++ {
-		job, err := p.K8sClient.GetJob(jobName, jfsConfig.Namespace)
+	for {
+		job, err := p.K8sClient.GetJob(ctx, jobName, jfsConfig.Namespace)
 		if err != nil {
 			if k8serrors.IsNotFound(err) {
 				klog.Infof("waitUtilJobCompleted: Job %s is completed and been recycled", jobName)
 				return nil
 			}
+			if k8serrors.IsTimeout(err) {
```

Comment:
ctx may not have timeout specified, so this will change the behavior (not expected).

Reply:
Changes reverted